sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpAnnotatedMessageExtensions.cs

```diff
@@ -399,9 +399,9 @@ amqpBody switch
         private static void CopyDictionary<TKey, TValue>(IDictionary<TKey, TValue> source,
                                                          IDictionary<TKey, TValue> destination)
         {
-            foreach (var key in source.Keys)
+            foreach (var pair in source)
```

Comment:
My also check `if (source.Count > 0)` before allocating enumerator

Reply:
I thought of that and was hesitant to introduce branching overhead with the scenario heavily weighted to cases where there will be items.   However, your poke made me curious enough to give it a look.   Synthetic benchmarks show minimal impact for introducing the `if` and a decent savings in the off chance of empty.   Since the change is trivial and doesn't add additional complexity, I included it.

Simple benchmark results using 10, 100, 1,000, 10,000 item sets:
```text

|        Method |          Mean |         Error |        StdDev |    Gen 0 |    Gen 1 |    Gen 2 | Allocated |
|-------------- |--------------:|--------------:|--------------:|---------:|---------:|---------:|----------:|
|  NoCheckEmpty |      35.59 ns |      0.733 ns |      0.784 ns |   0.0162 |        - |        - |     136 B |
|    CheckEmpty |      16.36 ns |      0.246 ns |      0.230 ns |   0.0095 |        - |        - |      80 B |
|  NoCheckSmall |     576.58 ns |      6.083 ns |      5.392 ns |   0.1249 |        - |        - |    1048 B |
|    CheckSmall |     583.01 ns |      7.204 ns |      6.386 ns |   0.1249 |        - |        - |    1048 B |
| NoCheckMedium |   5,248.85 ns |     89.378 ns |     83.604 ns |   1.2207 |   0.0305 |        - |   10248 B |
|   CheckMedium |   5,275.98 ns |     74.961 ns |     70.118 ns |   1.2207 |   0.0305 |        - |   10248 B |
|  NoCheckLarge |  49,208.16 ns |    798.558 ns |    746.971 ns |  12.1460 |   2.3804 |        - |  102272 B |
|    CheckLarge |  50,063.35 ns |    812.321 ns |    720.102 ns |  12.1460 |   2.3804 |        - |  102272 B |
|   NoCheckHuge | 914,093.58 ns | 10,112.536 ns |  9,459.272 ns | 221.6797 | 221.6797 | 221.6797 |  942320 B |
|     CheckHuge | 901,393.36 ns | 11,549.407 ns | 10,803.322 ns | 221.6797 | 221.6797 | 221.6797 |  942320 B |
```